deno_typescript/lib.rs

```diff
@@ -204,7 +204,12 @@ pub fn mksnapshot_bundle(
   js_check(
     isolate.execute(&bundle_filename.to_string_lossy(), bundle_source_code),
   );
-  let script = &format!("__instantiate(\"{}\");", main_module_name);
+  // instantiate the bundle and delete __proto__ for security reasons
+  // this is intentionally not compliant with ECMA-262 Annex B.2.2.1.
+  let script = &format!(
+    "__instantiate(\"{}\");\n\ndelete Object.prototype.__proto__;",
+    main_module_name
+  );
```

Comment:
I wonder if deno_typescript is the appropriate place for this. Can’t it be done during startup in CLI?

Reply:
That was the first version, then @bartlomieju suggested it be done before the snapshot is taken.  I can see a benefit of it being part of the snapshot, but I guess it should be an option for `deno_typescript`?

Reply:
Maybe it should live in `cli/js/globals.ts`? That file is loaded for all entry points for snapshotting

Reply:
Yes, but we don't have any code that automatically runs when that module loads, so I think that would be surprising.  I have moved it to `main.ts` and `compiler.ts` to run before the bootstrapping functions are placed in the global scope.